src/server/server.ts

```diff
@@ -110,16 +115,22 @@ export default class Server implements Hub.RouteBuilder {
 
     this.route("/actions/:actionId/execute", async (req, res) => {
       const request = Hub.ActionRequest.fromRequest(req)
-      const action = await Hub.findAction(req.params.actionId, {lookerVersion: request.lookerVersion})
-      const actionResponse = await action.validateAndExecute(request)
-
-      // Some versions of Looker do not look at the "success" value in the response
-      // if the action returns a 200 status code, even though the Action API specs otherwise.
-      // So we force a non-200 status code as a workaround.
-      if (!actionResponse.success) {
-        res.status(400)
+      const action = await Hub.findAction(req.params.actionId, { lookerVersion: request.lookerVersion })
```

Comment:
Can we move all the queue management stuff into its own file/class so that it's not all intermixed with the server code?

Reply:
That way we can make the interface to calling the action cross process just look like a single async function call